### ISSUE TITLE
TextDocumentContentChangeEvent requires `range` while it is optional in LSP spec

### ISSUE THREAD

**Issue by soutaro** [mtsmfm/language_server-protocol-ruby#48]
The `range` keyword of `TextDocumentContentChangeEvent` is required in version 3.17.0, but the LSP spec allows having only `text` attribute.

* https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocumentContentChangeEvent
* https://github.com/mtsmfm/language_server-protocol-ruby/blob/main/lib/language_server/protocol/interface/text_document_content_change_event.rb#L9

The best method signature would be the following:

```rb
# text is mandatory because all of the interfaces requires the attribute.
# range and range_length are optional even if one of the interfaces requires it.
def initialize(range: nil, range_length: nil, text:)
```

@paracycle @mtsmfm

**Comment by mtsmfm** [comment]
@soutaro It seems we have two problems

1. New generator doesn't handle union type well
2. [Meta model](https://github.com/microsoft/language-server-protocol/blob/gh-pages/_specifications/lsp/3.17/specification.md#metaModel) and Markdown has some inconsistency

I restored old (TS based) generator and I'm checking the difference in #49
I would be happy if you help to review it

**Comment by soutaro** [comment]
Thank you @mtsmfm. 🙏 
Looks like the old generator is doing correctly (as I expected.) An attribute is required only when all of the union clauses requires it.

* https://github.com/mtsmfm/language_server-protocol-ruby/pull/49/files#diff-350ffa9b1c3c1b8ca1bef8e209ad155a7772e060ca2e5f75ea848aa919a15a55R171

**Comment by mtsmfm** [comment]
@soutaro I've released 3.17.0.1 which contains the fix. Thank you for reporting 🙏
